command.go

```diff
@@ -89,10 +96,46 @@ func (c *Command) FullName() string {
 	return strings.Join(c.commandNamePath, " ")
 }
 
-// Run invokes the command given the context, parses ctx.Args() to generate command-specific flags
-func (c *Command) Run(ctx *Context) (err error) {
-	if len(c.Subcommands) > 0 {
-		return c.startApp(ctx)
+func (cmd *Command) Command(name string) *Command {
+	for _, c := range cmd.Subcommands {
+		if c.HasName(name) {
+			return c
+		}
+	}
+
+	return nil
+}
+
+func newRootCommand(a *App) *Command {
+	return &Command{
+		HelpName:               a.HelpName,
+		Subcommands:            a.Commands,
+		flagCategories:         a.flagCategories,
+		Flags:                  a.Flags,
+		Name:                   a.Name,
+		Action:                 a.Action,
+		UseShortOptionHandling: a.UseShortOptionHandling,
+		Before:                 a.Before,
+		After:                  a.After,
+		HideHelp:               a.HideHelp,
+		HideHelpCommand:        a.HideHelpCommand,
+		OnUsageError:           a.OnUsageError,
+		CustomHelpTemplate:     a.CustomAppHelpTemplate,
+		Usage:                  a.Usage,
+		UsageText:              a.UsageText,
+		Description:            a.Description,
+		ArgsUsage:              a.ArgsUsage,
+		BashComplete:           a.BashComplete,
+		categories:             a.categories,
+		isRoot:                 true,
```

Comment:
This is very much a nit, but can these be sorted?

Reply:
Sorted in same order as definition